cloud/google/serviceaccount.go

```diff
@@ -127,12 +139,9 @@ func (gce *GCEClient) createSecretForServiceAccountKey(accountId string, project
 
 // creates a service account with the roles specifed. Returns the account id
 // of the created account and the project it belongs to.
-func (gce *GCEClient) createServiceAccount(serviceAccountPrefix string, roles []string, cluster *clusterv1.Cluster, initialMachines []*clusterv1.Machine) (string, string, error) {
-	if len(initialMachines) == 0 {
```

Comment:
No longer needed to check the number of initial machines? What was the original reason then? Or, is it checked elsewhere?

Reply:
This actually should have been removed in a previous PR (#183), it was an oversight on my part. The project field used to be on the machine's provider config, so we had to make sure there was at least 1 machine that we could grab the project from. Once it was moved to the cluster this was no longer necessary because we can get that value from the cluster.